src/lib/dom-api-mutation.html

```diff
@@ -0,0 +1,280 @@
+            options.attributes;
+          if (!this._observer) {
+            this._observer = new MutationObserver(this._notify.bind(this));
+            this._preflush = this._flush.bind(this);
+            Polymer.dom.addPreflush(this._preflush);
+            // note: doing this > 1x is a no-op
+            this._observer.observe(this.node, {childList: true});
+          }
+          ensureSetup.call(this, options);
+        },
+
+        _ensureCleanup: function() {
+          this._observer.disconnect();
+          Polymer.dom.removePreflush(this._preflush);
+        },
+
+        _notify: function(mxns) {
+          var info = {
+            target: this.node,
+            addedNodes: [],
+            removedNodes: []
+          };
+          // collapse multiple mutations into one view
+          if (Array.isArray(mxns)) {
+            Array.prototype.forEach.call(mxns, function(m) {
+              if (m.addedNodes) {
+                for (var i=0; i < m.addedNodes.length; i++) {
+                  info.addedNodes.push(m.addedNodes[i]);
+                }
+              }
+              if (m.removedNodes) {
+                for (var i=0; i < m.removedNodes.length; i++) {
+                  info.removedNodes.push(m.removedNodes[i]);
+                }
+              }
+            });
+          } else if (mxns) {
+            info = mxns;
+          }
+          if (info.addedNodes.length || info.removedNodes.length) {
+            this._updateContentElements(info);
+            // attribute tracking helps us notice distributed nodes changes
+            // needed only with shadow dom.
+            if (this._trackAttributes) {
+              this._updateAttributeObservers(info);
+            }
+            this._callListeners(info);
+          }
+        },
+
+        _flush: function() {
+          if (this._observer) {
+            this._notify(this._observer.takeRecords());
+          }
+        },
+
+        _observeContent: function(content) {
+          return Polymer.dom(content).observeNodes(this._notify.bind(this), {
+            attributes: this._trackAttributes
+          });
+        },
+
+        _updateAttributeObservers: function(info) {
+          this._observeAttributes(info.addedNodes);
+          this._unobserveAttributes(info.removedNodes);
+        },
+
+        _observeAttributes: function(elements) {
+          for (var i=0, h, n; (i < elements.length) && (n=elements[i]); i++) {
+            if (n.nodeType === Node.ELEMENT_NODE) {
+              this._ensureElementAttrObserver(n);
+            }
+          }
+        },
+
+        _ensureElementAttrObserver: function(element) {
+          this._attrObservers = this._attrObservers || new WeakMap();
+          if (!this._attrObservers.get(element)) {
+            var self = this;
+            this._attrObservers.set(element, new MutationObserver(
+              function(mxns) {
+                self._callListeners(mxns, function(options) {

```

Comment:
These are attribute mutations (not add/remove mutations)... Ideally needs to be converted to addedNodes/removedNodes.